src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java

```diff
@@ -96,25 +97,70 @@ public static AminoAcid getEukaryoticAminoAcidByCodon(final String codon) {
+    public static AminoAcid getMitochondrialAminoAcidByCodon(final String rawCodon) {
+        return getMitochondrialAminoAcidByCodon(rawCodon, false, false, false, false, false);
+    }
 
-        if (codon == null) {
+    /**
+     * Returns the {@link AminoAcid} corresponding to the given three-letter Mitochondrial {@code rawCodon}.
+     * The codons given are expected to be valid for Mitochondrial DNA.
+     * @param rawCodon The three-letter codon (each letter one of A,[T or U],G,C) representing a Mitochondrial {@link AminoAcid}
+     * @param isGenusBosAndFirst {@code True} iff the codon to be decoded is from a sample in the genus Bos (cow) and came first in the coding sequence.
+     * @param isGenusHomoAndFirst {@code True} iff  the codon to be decoded is from a sample in the genus Homo (human) and came first in the coding sequence.
+     * @param isGenusMusAndFirst {@code True} iff  the codon to be decoded is from a sample in the genus Mus (mouse) and came first in the coding sequence.
+     * @param isGenusCorturnixAndFirst {@code True} iff  the codon to be decoded is from a sample in the genus Corturnix (quail) and came first in the coding sequence.
+     * @param isGenusGallusAndFirst {@code True} iff  the codon to be decoded is from a sample in the genus Gallus (chicken) and came first in the coding sequence.
+     * @return The {@link AminoAcid} corresponding to the given {@code rawCodon}.  Returns {@code null} if the given {@code rawCodon} does not code for a Mitochondrial {@link AminoAcid}.
+     */
+    public static AminoAcid getMitochondrialAminoAcidByCodon(final String rawCodon,
+                                                             final boolean isGenusBosAndFirst,
+                                                             final boolean isGenusHomoAndFirst,
+                                                             final boolean isGenusMusAndFirst,
+                                                             final boolean isGenusCorturnixAndFirst,
+                                                             final boolean isGenusGallusAndFirst) {
+        if (rawCodon == null) {
             return null;
         }
 
-        final String upperCodon = codon.toUpperCase();
-        if ( isFirst && upperCodon.equals("ATT") || upperCodon.equals("ATA") ) {
+        // TODO: Need to solicit more info on partly coded stop codons as alluded to here: https://www.sciencedirect.com/science/article/pii/S0005272898001613
+
+        // Convert Uracils to Thymines and convert to upper case so we can use our normal lookup table:
+        // Note: this may be unnecessary, but is here for safety and correctness (at least according to the
+        //       lookup table on wikipedia https://en.wikipedia.org/wiki/Vertebrate_mitochondrial_code).
+        final String upperCodon = rawCodon.replaceAll("[Uu]", "T").toUpperCase();
```

Comment:
Can't you use a pre-populated map instead of a bunch of if statements?

Reply:
Yes, but I didn't want to make another map of codons to amino acids since the codes are so similar.

I also wanted to make it explicit and clear that these differences occur in genus-specific ways.

I will add a comment and use a `HashMap`, but if somehow the comment gets deleted then this will get confusing.